language/tools/move-cli/src/main.rs

```diff
@@ -213,9 +214,9 @@ fn publish(args: &Move, files: &[String]) -> Result<OnDiskStateView> {
     if args.verbose {
         println!("Compiling Move modules...")
     }
-    let (_, compiled_units) =
+    let (files, compiled_units_res) =
         move_lang::move_compile(files, &args.get_compilation_deps()?, None, None)?;
-
+    let compiled_units = move_lang::unwrap_or_report_errors!(files, compiled_units_res);
```

Comment:
It seems like `move_compile` followed by `unwrap_or_report_errors` is the most common pattern. Perhaps we could expose a `move_compile_and_report` (or similar) API that does these two steps for you like the old `move_compile` API did?

Reply:
I was trying to reduce the number of entry points, but maybe you are right...

Reply:
At the very least I like the more positive phrasing of `_and_report` than the old negative `_no_report`, makes the functions a bit cleaner to define too